app/views/shared/_run_table.slim

```diff
@@ -43,8 +43,8 @@
             - if cols.include?(:rival)
               td.align-right
                 small
-                  - rival = run.user.rival_for(run.category)
-                  - if rival.present? && rival.runs?(run.category)
+                  - rival = run.user.rivals.detect { |r| r.category_id == run.category_id }
```

Comment:
Is this better because it caches the `run.user.rivals` query for use in all table rows?

Reply:
Yep, it caches the rivals call for every loop.  I'm not sure if it caches the array that it turns it into though, so if you wanted maximum performance you could call rivals once at the top, and then use that local variable for each loop iteration.

This still requires an extra call to get the rivals run, just like before, but we eliminate that first call for every run.